core/tests/test_chunk.h

```diff
@@ -0,0 +1,156 @@
+    TEST_CHECK(chunk_paint_block(a, 5, 5, 5, 2) == true);
+    TEST_CHECK(chunk_paint_block(a, 6, 6, 6, 3) == true);
+    TEST_CHECK(chunk_paint_block(a, 0, 0, 0, 0) == false);
+
+    // chunk_get_block()
+    // Check if the block is placed at the right spot in the chunk
+    // Also check if the previous function of paint worked
+    Block *check = chunk_get_block(a, 4, 4, 4);
+    TEST_CHECK(check->colorIndex == 1);
+    check = chunk_get_block(a, 5, 5, 5);
+    TEST_CHECK(check->colorIndex == 2);
+    check = chunk_get_block(a, 6, 6, 6);
+    TEST_CHECK(check->colorIndex == 3);
+    check = chunk_get_block(a, 11, 9, 6);
+    TEST_CHECK(check == NULL);
+
+    // chunk_get_block_2()
+    int3 pos = {4, 4, 4};
+    check = chunk_get_block_2(a, &pos);
+    TEST_CHECK(check->colorIndex == 1);
+    int3_set(&pos, 5, 5, 5);
+    check = chunk_get_block_2(a, &pos);
+    TEST_CHECK(check->colorIndex == 2);
+    int3_set(&pos, 6, 6, 6);
+    check = chunk_get_block_2(a, &pos);
+    TEST_CHECK(check->colorIndex == 3);
+    int3_set(&pos, 11, 9, 6);
+    check = chunk_get_block_2(a, &pos);
+    TEST_CHECK(check == NULL);
+
+    int NBBlocks = chunk_get_nb_blocks(a);
+    TEST_CHECK(NBBlocks == 3);
+
+    // chunk_get_block_pos()
+    chunk_get_block_pos(a, 4, 4, 4, &pos);
+    TEST_CHECK(pos.x == 14);
+    TEST_CHECK(pos.y == 12);
+    TEST_CHECK(pos.z == 9);
+    chunk_get_block_pos(a, 5, 5, 5, &pos);
+    TEST_CHECK(pos.x == 15);
+    TEST_CHECK(pos.y == 13);
+    TEST_CHECK(pos.z == 10);
+    chunk_get_block_pos(a, 6, 6, 6, &pos);
+    TEST_CHECK(pos.x == 16);
+    TEST_CHECK(pos.y == 14);
+    TEST_CHECK(pos.z == 11);
+
+    // chunk_get_inner_bounds()
+    CHUNK_COORDS_INT_T minX = 0;
+    CHUNK_COORDS_INT_T minY = 0;
+    CHUNK_COORDS_INT_T minZ = 0;
+    CHUNK_COORDS_INT_T maxX = 0;
+    CHUNK_COORDS_INT_T maxY = 0;
+    CHUNK_COORDS_INT_T maxZ = 0;
+    chunk_get_inner_bounds(a, &minX, &maxX, &minY, &maxY, &minZ, &maxZ);
+    TEST_CHECK(minX == 4 && minY == 4 && minZ == 4 && maxX == 7 && maxY == 7 && maxZ == 7);
+
+    // chunk_removeBlock()
+    TEST_CHECK(chunk_removeBlock(a, 4, 4, 4) == true);
+    check = chunk_get_block(a, 4, 4, 4);
+    TEST_CHECK(check == NULL);
+    NBBlocks = chunk_get_nb_blocks(a);
+    TEST_CHECK(NBBlocks == 2);
+    TEST_CHECK(chunk_removeBlock(a, 5, 5, 5) == true);
+    check = chunk_get_block(a, 5, 5, 5);
+    TEST_CHECK(check == NULL);
+    NBBlocks = chunk_get_nb_blocks(a);
+    TEST_CHECK(NBBlocks == 1);
+    TEST_CHECK(chunk_removeBlock(a, 6, 6, 6) == true);
+    check = chunk_get_block(a, 6, 6, 6);
+    TEST_CHECK(check == NULL);
+    NBBlocks = chunk_get_nb_blocks(a);
+    TEST_CHECK(NBBlocks == 0);
+    TEST_CHECK(chunk_removeBlock(a, 0, 0, 0) == false);
+
+    chunk_destroy(a);
+}
+
+// Create a chunk and set diffrents values on the "display bool" of this chunk.
+// Then check if the bool is set with the good values
+void test_chunk_needs_display(void) {
+    Chunk *a = chunk_new(10, 10, 10);
```

Comment:
A better variable name could be find for the chunk here, like c or chunk.
Also, it's probably best to use different values for the 3 dimensions x, y, z.